server.go

```diff
@@ -88,6 +91,50 @@ func RedirectLegacyExamples() {
 	}
 }
 
+func submitFormXHR(w http.ResponseWriter, r *http.Request){
+	name := r.FormValue("name")
+	w.Header().Add("AMP-Access-Control-Allow-Source-Origin", buildSourceOrigin(r.Host))
+	w.Header().Add("Content-Type", "application/json")
+	if name == "error" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("{\"name\":\"%s\"}", name)))		
+	} else {
+		var sourceOrigin bytes.Buffer
+		if strings.HasPrefix(r.Host, "localhost") {
+			sourceOrigin.WriteString("http://")
+		} else {
+			sourceOrigin.WriteString("https://")
+		}
+		sourceOrigin.WriteString(r.Host)	
+  	email := r.FormValue("email")
+  	w.Write([]byte(fmt.Sprintf("{\"name\":\"%s\", \"email\":\"%s\"}", name, email)))
+  }		
+}
+
+func submitForm(w http.ResponseWriter, r *http.Request){
+	name := r.FormValue("name")
+	w.Header().Add("AMP-Access-Control-Allow-Source-Origin", buildSourceOrigin(r.Host))
+
+	if name == "error" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("Oops! We apologise something went wrong. Please try again later...thanks %s for trying amp-form demo with an error response.", name)))		
+	} else {		
+  	email := r.FormValue("email")
+  	w.Write([]byte(fmt.Sprintf(" Success! Thanks %s for trying amp-form demo! This is just a demo, so you will not get any email at the email %s that you provided! Try to insert the word \"error\" as a name input in the form to see how amp-form supports errors.", name, email)))
+  }		
+}
+
+func buildSourceOrigin(host string) string{
```

Comment:
Because localhost is not gonna be on https and I would get the error:
log.js:350 Uncaught Error: Form submission failed:: Returned AMP-Access-Control-Allow-Source-Origin is not equal to the current: https://localhost:8080 vs http://localhost:8080​​​